include/s3select_json_parser.h

```diff
@@ -563,8 +576,10 @@ class JsonParserHandler : public rapidjson::BaseReaderHandler<rapidjson::UTF8<>,
 	json_key_value_t key_value(key_path,v);
 	m_star_operation_cb(key_value);
       }
-	if (prefix_match)
-      		variable_match_operations.new_value(v);
+	if (prefix_match) {
+          json_key_value_t key_path_value(key_path,v);
```

Comment:
`key_path_value` should populate only upon `json-format-output` is true.
no need to construct `key_path_value` each time.